app/components/post_priority/post_priority_label.tsx

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useIntl} from 'react-intl';
+import {StyleProp, StyleSheet, Text, View, ViewStyle} from 'react-native';
+
+import CompassIcon from '@components/compass_icon';
+import {PostPriorityTypes} from '@constants/post';
+import {typography} from '@utils/typography';
+
+const style = StyleSheet.create({
+    container: {
+        flexDirection: 'row',
+        borderRadius: 4,
+        alignItems: 'center',
+        height: 16,
+        paddingHorizontal: 4,
+    },
+    label: {
+        color: '#fff',
+        ...typography('Body', 25, 'SemiBold'),
+    },
+    icon: {
+        color: '#fff',
+        fontSize: 12,
+        marginRight: 4,
+    },
+});
+
+type Props = {
+    label: string;
+};
+
+const PostPriorityLabel = ({label}: Props) => {
+    const intl = useIntl();
+
+    const containerStyle: StyleProp<ViewStyle> = [style.container];
+    let iconName = '';
+    let labelText = '';
+    if (label === PostPriorityTypes.URGENT) {
+        containerStyle.push({backgroundColor: '#D24B4E'});
```

Comment:
this colors are hardcoded?

Also if the case is to have hardcoded colors, why not define them in the styles and just do `containerStyle.push(style.urgent)` etc..

Reply:
Yes, colors are hardcoded. Moving them to styles. I was in two minds about whether to have it like this or in the styles 😄